chingu-voyages/v52-tier3-team-38
# Ross/feature/resident profile page

# Description
Creating another pull request before I get into even more scope creep (if I haven't already) or get lost. A simple profile page that fetches user data and appointment data. This logic can then be re-used in a bigger way in Service History.

Users should be able to view all of their appointments, and the status (cancelled or pending). Appointment statuses should update from "pending" to "canceled" when clicking Cancel button. I tried to implement logic that only pulls appointments related to appointment ID tied to user ID. Hard to see if it works, because I get an error when I try to create an appointment on Submit Request page.

## Related to the following Jira/Trello issue 

- Resident Interface
- List view of resident appointments.

## Main changes explained:
- Created src/app/user/[_id]/profile/page.js to implement user profile interface
- Updated src/app/api/appointments/route.js GET endpoint to filter appointments by user ID
- Updated src/app/api/appointments/route.js PUT endpoint to handle appointment cancellation
- Integrated with existing user authentication and navigation system

## How to test:
1. check into current branch
2. Check into Ross/feature/resident-profile-page branch
3. Run npm install and npm run dev.
4. Open app at localhost:3000 and log in (if not already) 
5. Go to profile page.
6. Verify user information (name and email) is displayed at top of scrreen.
7. Verify there are no appointments displayed at bottom of screen.

## Screenshots or videos of changes:
![image](https://github.com/user-attachments/assets/7c2de1f0-7b03-4a17-9546-f607bd9249a4)


## Note:
- Once I can submit and fetch appointments, I'd like to set a limit for THIS page to no more than 2 pending appointment cards showing at a time. Cancelled appointments and most other appointments can be looked at in the Service History page. I'd like to try to refine the logic in the Service History page, and see what I can then re-use BACK in Profile page.
- Basic Bootstrap styling included in footer navbar and page. I like card components. Turns out in bootstrap stuff is "fixed", NOT "sticky".
